OpenEdXMobile/src/main/java/org/edx/mobile/util/CourseDateUtil.kt

```diff
@@ -0,0 +1,47 @@
+package org.edx.mobile.util
+
+import android.content.Context
+import android.text.TextUtils
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import org.edx.mobile.R
+import org.edx.mobile.model.course.CourseBannerInfoModel
+import org.edx.mobile.model.course.CourseBannerType
+
+/**
+ * Utils class for CourseDates
+ */
+class CourseDateUtil {
+    companion object {
+
+        @JvmStatic
+        fun getCourseDatesBanner(view: View, courseBannerInfoModel: CourseBannerInfoModel, clickListener: View.OnClickListener) {
+            val context: Context = view.context
+            val textView: TextView = view.findViewById(R.id.banner_info)
+            val button: Button = view.findViewById(R.id.btn_shift_dates)
+            var buttonText = ""
+            // Currently we are only handling RESET_DATES case,
+            // TODO UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
+            when (courseBannerInfoModel.datesBannerInfo.getCourseBannerType()) {
+                CourseBannerType.UPGRADE_TO_GRADED -> textView.text = context.getText(R.string.course_dates_banner_upgrade_to_graded)
+                CourseBannerType.UPGRADE_TO_RESET -> textView.text = context.getText(R.string.course_dates_banner_upgrade_to_reset)
+                CourseBannerType.RESET_DATES -> {
+                    textView.text = context.getText(R.string.course_dates_banner_reset_date)
+                    buttonText = context.getString(R.string.course_dates_banner_reset_date_button)
+                }
+                CourseBannerType.INFO_BANNER -> textView.text = context.getText(R.string.course_dates_info_banner)
+                CourseBannerType.BLANK -> view.visibility = View.GONE
+            }
+
+            if (!TextUtils.isEmpty(textView.text)) {
+                if (!TextUtils.isEmpty(buttonText)) {
+                    button.text = buttonText
+                    button.visibility = View.VISIBLE
+                    button.setOnClickListener(clickListener)
+                }
+                view.visibility = View.VISIBLE
+            }
```

Comment:
also need to check the buttonText is empty, cuz button should be hide after shifting the dates.

```java
        if (!TextUtils.isEmpty(textView.getText())) {
            if (!TextUtils.isEmpty(buttonText)) {
                button.setText(buttonText);
                button.setOnClickListener(v -> {
                    courseDateViewModel.resetCourseDatesBanner(unit.getCourseId());
                });
            }
            infoBanner.setVisibility(View.VISIBLE);
        }
        button.setVisibility(TextUtils.isEmpty(buttonText) ? View.GONE : View.VISIBLE);
```
**e.g**
<img src="https://user-images.githubusercontent.com/43750646/98087241-9d25cc00-1ea1-11eb-9edb-b57c75a2763a.gif" height="420" />